modules/ROOT/pages/6.0-release-notes-apis.adoc

```diff
@@ -0,0 +1,209 @@
+
+It also documents the {productname} 6.0 APIs that have been removed.
+
+
+* xref:apis[APIs]
+* xref:new-and-improved-apis[New and improved APIs]
+* xref:changed-apis[Changed APIs]
+* xref:fixed-apis[Fixed APIs]
+* xref:removed-or-deprecated-apis[Removed or deprecated APIs]
+
+// tag::apis[]
+[[apis]]
+== APIs
+
+[[new-and-improved-apis]]
+=== New and improved APIs
+
+[[editor-annotator-removeall]]
+==== `editor.annotator.removeAll`
+
+The new `editor.annotator.removeAll` API replaces the `removeAll` function built-in to the Comments plugin.
+
+[[editor-options]]
+==== `editor.options` and `tinymce.activeEditor.options`
+
+The new `editor.options` API replaces two old APIs: `editor.settings` and `editor.getParam`.
+
+Along with this new API, the new `tinymce.activeEditor.options` registry replaces `tinymce.activeeditor.settings`.
+
+[[editor-uploadImages]]
+==== `editor.uploadImages()`
+
+The `editor.uploadImages()` API now includes a `remove` flag.
+
+The flag returns `true` when an image being uploaded is removed after failing to upload.
+
+[[fakeclipboard]]
+==== `FakeClipboard`
+
+The new `FakeClipboard` API provides a means for storing state that can be set and accessed by plugins and the core editor.
+
+`FakeClipboard` is implemented on the {productname} global, allowing anything to write to it and read from it.
+
+[[resource-unload]]
+==== `Resource.unload`
+
+The new `Resource.unload` API makes it possible to unload resources in the resource loader.
+
+This allows tests, such as `ImageEmojiTest`, to remove existing databases before loading their own.
+
+[[setdata]]
+==== `setData` method in dialog API
+
+The dialog API’s `setData` method now uses a deep merge algorithm.
+
+Consequently, it now supports partial nested objects.
+
+[[promise]]
+==== `success` and `failure` callbacks replaced by `Promise` in multiple APIs
+
+ Previously, the AddOnManager, PluginManager, ScriptLoader, StyleSheetLoader, and ThemeManager APIs report resource loading status with `success` or `failure` callbacks.
+
+These APIs now, instead, return a `Promise`, with a value of either `resolve` or `reject`. Also, `EditorUpload`, which already returns a `Promise`, had callbacks removed.
+
+[[changed-apis]]
+=== Changed APIs
+
+[[closebutton]]
+==== `closeButton` and `timeout`
+
+* Previously, using `closeButton` to disable a _Notification’s_ close button required a `timeout` value to also be set.
++
+In {productname} 6, `closeButton` shows or hides the close button regardless of a `timeout` value being set or not set.
++
+Equivalently, setting a `timeout` value sets how long the notification presents before automatically closing, regardless of whether a close button is displayed or not.
+
+[[disabled]]
+==== `disabled` functions renamed in all {productname} APIs
+
+Previously, multiple {productname} APIs and UI classes used `disable` or `disabled` in their function or property names.
```

Comment:
`classes` seems like the wrong term here? In this case we probably should say `UI components` instead.

```suggestion
Previously, multiple {productname} APIs and UI components used `disable` or `disabled` in their function or property names.
```